src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -250,23 +250,15 @@ bool InstanceKlass::has_as_permitted_subclass(const InstanceKlass* k) const {
     return false;
   }
 
-  // Check for a resolved cp entry, else fall back to a name check.
+  // Do not check for a resolved cp entry, because that check can fail if
+  // the class is being redefined.  Just do a name check.
   // We don't want to resolve any class other than the one being checked.
```

Comment:
You don't resolve any classes at all now so this comment can go.

Reply:
comment was deleted.